cpp/src/aztec/honk/composer/composer_helper/permutation_helper.hpp

```diff
@@ -175,4 +175,50 @@ void compute_standard_honk_sigma_permutations(CircuitConstructor& circuit_constr
     }
 }
 
+/**
+ * @brief Compute standard honk id polynomials and put them into cache
+ *
+ * @details Honk permutations involve using id and sigma polynomials to generate variable cycles. This function
+ * generates the id polynomials and puts them into polynomial cache, so that they can be used by the prover.
+ *
+ * @tparam program_width The number of witness polynomials
+ * @param key Proving key where we will save the polynomials
+ */
+template <size_t program_width> void compute_standard_honk_id_polynomials(proving_key* key)
+{
+    const size_t n = key->n;
+    // Fill id polynomials with default values
+    std::vector<barretenberg::polynomial> id_polynomials_lagrange;
+    for (uint64_t i = 0; i < program_width; ++i) {
+        // Construct permutation polynomials in lagrange base
+        std::string index = std::to_string(i + 1);
+        id_polynomials_lagrange.push_back(barretenberg::polynomial(key->n));
+        barretenberg::polynomial& id_polynomial_lagrange = id_polynomials_lagrange[i];
+        for (uint64_t j = 0; j < key->n; j++) {
+            id_polynomial_lagrange[j] = (i * n + j);
+        }
+    }
+    // Save to polynomial cache
+    for (size_t i = 0; i < program_width; i++) {
+        std::string index = std::to_string(i + 1);
+        key->polynomial_cache.put("id_" + index + "_lagrange", std::move(id_polynomials_lagrange[i]));
+    }
+}
+
+/**
+ * @brief Compute Lagrange Polynomials L_0 and L_{n-1} and put them in the polynomial cache
+ *
+ * @param key Proving key where we will save the polynomials
+ */
+void compute_edge_lagrange_polynomials(proving_key* key)
```

Comment:
I have two requests for this function: 1) Change the name to avoid the term 'edge'. We've used that term to refer to an 'edge' of the hypercube and I thought at first glance that this function had something to do with that - maybe `compute_first_and_last_lagrange_polynomials()`? 2) Add a simple test just for completeness even though the function is quite simple.